config/config.go

```diff
@@ -103,10 +104,26 @@ func loadForFileFromBytes(confBytes []byte, filePath string, kmsEncryptionContex
 	var rule *creationRule
 
 	for _, r := range conf.CreationRules {
-		if match, _ := regexp.MatchString(r.FilenameRegex, filePath); match {
+		if r.PathRegex == "" && r.FilenameRegex == "" {
 			rule = &r
 			break
 		}
+		if r.PathRegex != "" && r.FilenameRegex != "" {
+			return nil, fmt.Errorf("error loading config: both filename_regex and path_regex were found, use only path_regex")
+		}
+		if r.FilenameRegex != "" {
+			if match, _ := regexp.MatchString(r.FilenameRegex, filePath); match {
+				fmt.Println("WARNING: filename_regex will be removed in a future release, use path_regex instead")
```

Comment:
I think this error message could be improved, it should tell users what they should do. If I put myself in the shoes of a user that hasn't touched their `.sops.yaml` file in months, I feel very uncertain about what action I need to take and why.